### PR TITLE
[show] Add 'show interface neighbor expected' command

### PR DESCRIPTION
**Pull request by ivlee1**
**- What I did**
add a command to show interface neighbor metadata i.e. expected data

**- How I did it**
get 2 set of data from the database and output in table format

**- How to verify it**
run "show interface neighbor expected"

**- Previous command output (if the output of a command-line utility has changed)**
None 

**- New command output (if the output of a command-line utility has changed)**
```
admin@sonic:~$ show interfaces neighbor expected            
LocalPort    Neighbor    NeighborPort    NeighborLoopback    NeighborMgmt    NeighborType
-----------  ----------  --------------  ------------------  --------------  --------------
Ethernet112  ARISTA01T1  Ethernet1       None                172.16.140.124  LeafRouter
Ethernet116  ARISTA02T1  Ethernet1       None                172.16.140.125  LeafRouter
Ethernet120  ARISTA03T1  Ethernet1       None                172.16.140.126  LeafRouter
Ethernet124  ARISTA04T1  Ethernet1       None                172.16.140.127  LeafRouter
```
```
admin@sonic:~$ show interfaces neighbor expected Ethernet124 
LocalPort    Neighbor    NeighborPort    NeighborLoopback    NeighborMgmt    NeighborType
-----------  ----------  --------------  ------------------  --------------  --------------
Ethernet124  ARISTA04T1  Ethernet1       None                172.16.140.127  LeafRouter
```
```
admin@sonic"~$ show interfaces neighbor expected abcd       
LocalPort    Neighbor    NeighborPort    NeighborLoopback    NeighborMgmt    NeighborType
-----------  ----------  --------------  ------------------  --------------  --------------
```
-->

### CODE REVIEW

File: show/main.py [hunk 1]
```diff
@@ -228,6 +228,35 @@ def alias(interfacename):
...
+    neighborCmd = 'sonic-cfggen -d --var-json "DEVICE_NEIGHBOR"'
+    p1 = subprocess.Popen(neighborCmd, shell=True, stdout=subprocess.PIPE)
+    neighbor_dict = json.loads(p1.stdout.read())
+
+    neighborMetaDataCmd = 'sonic-cfggen -d --var-json "DEVICE_NEIGHBOR_METADATA"'
+    p2 = subprocess.Popen(neighborMetaDataCmd, shell=True, stdout=subprocess.PIPE)
+    neighborMetaData_dict = json.loads(p2.stdout.read())
+
+    #Swap Key and Value
+    tempDict= {}
```
Comment (jleveque) [inline show/main.py]: Add space before `=`

File: show/main.py [hunk 2]
```diff
@@ -228,6 +228,35 @@ def alias(interfacename):
 
     click.echo(tabulate(body, header))
 
+# 'Neighbor MetaData' subcommand ("show interfaces neighbor")
```
Comment (jleveque) [inline show/main.py]: This comment should simply notate the expected command that this function creates, like

`# 'neighbor' subcommand ("show interfaces neighbor")`

File: show/main.py [hunk 3]
```diff
@@ -228,6 +228,35 @@ def alias(interfacename):
...
+@interfaces.command()
+@click.argument('interfacename', required=False)
+def neighbor(interfacename):
+    """Show Interface Neighbor"""
```
Comment (jleveque) [inline show/main.py]: This docstring is used by the Click library as the help message for the command. Please make this more descriptive.

File: show/main.py [hunk 4]
```diff
@@ -228,6 +228,35 @@ def alias(interfacename):
...
+    neighbor_dict = json.loads(p1.stdout.read())
+
+    neighborMetaDataCmd = 'sonic-cfggen -d --var-json "DEVICE_NEIGHBOR_METADATA"'
+    p2 = subprocess.Popen(neighborMetaDataCmd, shell=True, stdout=subprocess.PIPE)
+    neighborMetaData_dict = json.loads(p2.stdout.read())
+
+    #Swap Key and Value
+    tempDict= {}
+    for port in natsorted(neighbor_dict.keys()):
+        tempDict[neighbor_dict[port]['name']]= {'localPort':port, 'neighborPort':neighbor_dict[port]['port']}
```
Comment (jleveque) [inline show/main.py]: Please add space before `=` and after colons,

`tempDict[neighbor_dict[port]['name']] = {'localPort': port, 'neighborPort': neighbor_dict[port]['port']}`

File: show/main.py [hunk 5]
```diff
@@ -228,6 +228,35 @@ def alias(interfacename):
...
+
+    header = ['LocalPort','Neighbor','NeighborPort','NeighborLoopback','NeighborMgmt','NeighborType']
```
Comment (jleveque) [inline show/main.py]: Please add space after each comma.

File: show/main.py [hunk 6]
```diff
@@ -228,6 +228,35 @@ def alias(interfacename):
...
+    body = []
+    if interfacename is not None:
+        for device in natsorted(neighborMetaData_dict.keys()):
+            if tempDict[device]['localPort']==interfacename:
```
Comment (jleveque) [inline show/main.py]: Add spaces around `==`

File: show/main.py [hunk 7]
```diff
@@ -228,6 +228,35 @@ def alias(interfacename):
...
+                body.append([ tempDict[device]['localPort'],device, tempDict[device]['neighborPort'], neighborMetaData_dict[device]['lo_addr'],neighborMetaData_dict[device]['mgmt_addr'],neighborMetaData_dict[device]['type'] ])
```
Comment (jleveque) [inline show/main.py]: Please ensure one space after each comma.

Also remove space after `[` and before `]`

Consider breaking long line.

File: show/main.py [hunk 8]
```diff
@@ -228,6 +228,35 @@ def alias(interfacename):
...
+    else:
+        for device in natsorted(neighborMetaData_dict.keys()):
+            body.append([ tempDict[device]['localPort'],device, tempDict[device]['neighborPort'], neighborMetaData_dict[device]['lo_addr'],neighborMetaData_dict[device]['mgmt_addr'],neighborMetaData_dict[device]['type'] ])
```
Comment (jleveque) [inline show/main.py]: Please ensure one space after each comma.

Also remove space after `[` and before `]`

Consider breaking long line.

**Comment by ivlee1** [comment]
Addressed all the comments

File: show/main.py [hunk 9]
```diff
@@ -228,6 +228,45 @@ def alias(interfacename):
 
     click.echo(tabulate(body, header))
 
+# 'neighbor' subcommand ("show interfaces neighbor")
+@interfaces.command()
+@click.argument('interfacename', required=False)
+def neighbor(interfacename):
+    """Show neighbor information by interfaces"""
+    neighborCmd = 'sonic-cfggen -d --var-json "DEVICE_NEIGHBOR"'
+    p1 = subprocess.Popen(neighborCmd, shell=True, stdout=subprocess.PIPE)
```
Comment (qiluo-msft) [inline show/main.py]: >p1 [](start = 4, length = 2)

Add some error handling?

File: show/main.py [hunk 10]
```diff
@@ -228,6 +228,45 @@ def alias(interfacename):
...
+    neighbor_dict = json.loads(p1.stdout.read())
+
+    neighborMetaDataCmd = 'sonic-cfggen -d --var-json "DEVICE_NEIGHBOR_METADATA"'
+    p2 = subprocess.Popen(neighborMetaDataCmd, shell=True, stdout=subprocess.PIPE)
```
Comment (qiluo-msft) [inline show/main.py]: >p2 [](start = 4, length = 2)

the same

File: show/main.py [hunk 11]
```diff
@@ -228,6 +228,45 @@ def alias(interfacename):
...
+    neighborMetaData_dict = json.loads(p2.stdout.read())
+
+    #Swap Key and Value
+    tempDict = {}
```
Comment (qiluo-msft) [inline show/main.py]: >tempDict [](start = 4, length = 8)

Give a better name?

File: show/main.py [hunk 12]
```diff
@@ -228,6 +228,45 @@ def alias(interfacename):
...
+    for port in natsorted(neighbor_dict.keys()):
+        tempDict[neighbor_dict[port]['name']] = {'localPort': port, 'neighborPort': neighbor_dict[port]['port']}
+
+    header = ['LocalPort', 'Neighbor', 'NeighborPort', 'NeighborLoopback', 'NeighborMgmt', 'NeighborType']
+    body = []
+    if interfacename is not None:
```
Comment (qiluo-msft) [inline show/main.py]: >if interfacename is not None: [](start = 4, length = 29)

if interfacename:

File: show/main.py [hunk 13]
```diff
@@ -228,6 +228,45 @@ def alias(interfacename):
 
     click.echo(tabulate(body, header))
 
+# 'neighbor' subcommand ("show interfaces neighbor")
```
Comment (qiluo-msft) [inline show/main.py]: >show interfaces neighbor [](start = 26, length = 24)

Suggest command line:
show interfaces neighbor expected.

In future, we may have other subcommand to observe the reality.
show interfaces neighbor arp
show interfaces neighbor lldp

File: show/main.py (hunk 9 above)
Comment (ivlee1) [reply show/main.py]: will add in the future for dynamic cmd

File: show/main.py (hunk 10 above)
Comment (ivlee1) [reply show/main.py]: will add in the future for dynamic cmd

File: show/main.py (hunk 12 above)
Comment (ivlee1) [reply show/main.py]: Fixed

File: show/main.py (hunk 13 above)
Comment (ivlee1) [reply show/main.py]: updated the cmd name as recommended

File: show/main.py [hunk 14]
```diff
@@ -228,6 +228,53 @@ def alias(interfacename):
...
 
+#
+# 'neighbor' group ###
+#
+@interfaces.group(cls=AliasedGroup, default_if_no_args=False)
+def neighbor():
+    """Show neighbor related information"""
+    pass
+
+# 'expected' subcommand ("show interface neighbor expected ")
```
Comment (qiluo-msft) [inline show/main.py]: >  [](start = 58, length = 1)

One extra blank?

**Pull request merged**